src/commands/containers/browseContainer.ts

```diff
@@ -0,0 +1,115 @@
+}
+
+// NOTE: These ports are ordered in order of preference.
+const commonWebPorts = [
+    443,    // SSL
+    80,     // HTTP
+    3000,   // (Node.js) Express.js
+    3001,   // (Node.js) Sails.js
+    5001,   // (.NET Core) ASP.NET SSL
+    5000,   // (.NET Core) ASP.NET HTTP
+    8080,   // (Node.js)
+    8081    // (Node.js)
+];
+
+const commonSslPorts = [
+    443,    // SSL
+    5001    // (.NET Core) ASP.NET SSL
+];
+
+function parsePortAndProtocol(portAndProtocol: string): { port: number, protocol: string | undefined } | undefined {
+    const splitPortAndProtocol = portAndProtocol.split('/');
+
+    if (splitPortAndProtocol.length > 0) {
+        const port = Number.parseInt(splitPortAndProtocol[0], 10);
+
+        if (!Number.isNaN(port)) {
+            return {
+                port,
+                protocol: splitPortAndProtocol.length > 1 ? splitPortAndProtocol[1].toLowerCase() : undefined
+            };
+        }
+    }
+
+    return undefined;
+}
+
+export async function browseContainer(context: IActionContext, node?: ContainerTreeItem): Promise<void> {
+    const telemetryProperties = <BrowseTelemetryProperties>context.telemetry.properties;
+
+    if (!node) {
+        node = await captureBrowseCancelStep('node', telemetryProperties, () => ext.containersTree.showTreeItemPicker<ContainerTreeItem>(ContainerTreeItem.runningContainerRegExp, context));
+    }
+
+    const inspectInfo = await node.getContainer().inspect();
+
+    const ports = inspectInfo && inspectInfo.NetworkSettings && inspectInfo.NetworkSettings.Ports || {};
+    const possiblePorts =
+        Object.keys(ports)
+              .map(portAndProtocol => ({ mappings: ports[portAndProtocol], containerPort: parsePortAndProtocol(portAndProtocol) }))
+              // Exclude port designations we cannot recognized...
+              .filter(port => port.containerPort !== undefined)
+              // Exclude ports that are non-TCP-based (which should be the default)...
+              .filter(port => port.containerPort.protocol === undefined || port.containerPort.protocol === 'tcp')
+              // Exclude ports not mapped to the host...
+              .filter(port => port.mappings && port.mappings.length > 0);
+
+    telemetryProperties.possiblePorts = possiblePorts.map(port => port.containerPort.port);
+
+    if (possiblePorts.length === 0) {
+        // tslint:disable-next-line: no-floating-promises
+        ext.ui.showWarningMessage('No valid ports were mapped from the container to the host.');
+
+        return;
+    }
+
+    let selectedPort =
+        possiblePorts.length === 1
+            // If there's just a single port, assume that one...
+            ? possiblePorts[0]
+            // Otherwise, prefer a common port (in order of preference)...
+            : commonWebPorts
+                .map(port => possiblePorts.find(possiblePort => port === possiblePort.containerPort.port))
+                .filter(possiblePort => possiblePort !== undefined)[0];
```

Comment:
Are sure the result of `filter()` will always have at least one element?

Reply:
`filter()` will always return a valid array, but it may be empty.  If that's the case, `[0]` will return undefined, which is ok in this scenario, as it will get checked in the next `if` statement.